controlplane/kubeadm/controllers/controller.go

```diff
@@ -482,6 +527,40 @@ func (r *KubeadmControlPlaneReconciler) reconcileHealth(ctx context.Context, clu
 	return ctrl.Result{}, nil
 }
 
+// healthCheck will analyse HealthCheckResult and report any errors discovered.
+// In addition to the health check, it also ensures there is a 1;1 match between nodes and machines.
+func healthCheck(controlPlane *internal.ControlPlane, nodeChecks internal.HealthCheckResult, clusterKey client.ObjectKey) error {
```

Comment:
nits, not blocking for this PR:
- clusterKey could be removed, given that we are passing the whole control place
- should this became a method of `internal.HealthCheckResult`, so this will read 
```
if err:= checkResult.Aggregate(controlPlane); err!=nil {
    ...
}
```

Reply:
This makes more sense, changed.